app/code/Magento/Wishlist/Test/Unit/Controller/Index/UpdateTest.php

```diff
@@ -0,0 +1,252 @@
+{
+    /**
+     * @var Validator $formKeyValidator
+     */
+    private $formKeyValidator;
+
+    /**
+     * @var WishlistProviderInterface $wishlistProvider
+     */
+    private $wishlistProvider;
+
+    /**
+     * @var LocaleQuantityProcessor $quantityProcessor
+     */
+    private $quantityProcessor;
+
+    /**
+     * @var Update $updateController
+     */
+    private $updateController;
+
+    /**
+     * @var $context
+     */
+    private $context;
+
+    /**
+     * @var Redirect $resultRedirect
+     */
+    private $resultRedirect;
+
+    /**
+     * @var ResultFactory $resultFatory
+     */
+    private $resultFactory;
+
+    /**
+     * @var RequestInterface $requestMock
+     */
+    private $requestMock;
+
+    /**
+     * @var ObjectManagerInterface $objectManagerMock
+     */
+    private $objectManagerMock;
+
+    /**
+     * @var ManagerInterface $messageManager
+     */
+    private $messageManager;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp()
+    {
+        $this->formKeyValidator = $this->createMock(Validator::class);
+        $this->wishlistProvider = $this->createMock(WishlistProviderInterface::class);
+        $this->quantityProcessor = $this->createMock(LocaleQuantityProcessor::class);
+        $this->context = $this->createMock(Context::class);
+        $this->resultRedirect = $this->createMock(Redirect::class);
+        $this->resultFactory = $this->createPartialMock(ResultFactory::class, ['create']);
+        $this->requestMock = $this->getMockBuilder(RequestInterface::class)
+            ->setMethods(['getPostValue'])
+            ->getMockForAbstractClass();
+        $this->objectManagerMock = $this->createMock(ObjectManagerInterface::class);
+
+        $this->context->expects($this->once())
+            ->method('getResultFactory')
+            ->willReturn($this->resultFactory);
+        $this->context->expects($this->once())
+            ->method('getObjectManager')
+            ->willReturn($this->objectManagerMock);
+
+        $this->resultFactory->expects($this->any())
+                              ->method('create')
+                              ->willReturn($this->resultRedirect);
+        $this->context->expects($this->any())
+            ->method('getRequest')
+            ->willReturn($this->requestMock);
+
+        $this->messageManager = $this->createMock(ManagerInterface::class);
+        $this->context->expects($this->any())
+            ->method('getMessageManager')
+            ->willReturn($this->messageManager);
+
+        $this->updateController = new Update(
+            $this->context,
+            $this->formKeyValidator,
+            $this->wishlistProvider,
+            $this->quantityProcessor
+        );
+    }
+
+    /**
+     * Test for update method Wishlist controller.
+     *
+     * @dataProvider getWishlistDataProvider
+     * @return void
+     */
+    public function testUpdate(array $wishlistDataProvider): void
```

Comment:
Instead of using `wishlistDataProvider['wishlist_data']` and `wishlistDataProvider['post_data']` just extract them as function parameters: `testUpdateForPostRequest(array $wishlistDataProvider, array $postData)` and introduce verification if no POST data provided: `testUpdateRedirectWhenNoPostData()`.